pystac_monty/sources/idu.py

```diff
@@ -0,0 +1,238 @@
+        return Collection.from_dict({})
+
+    def get_impact_collection(self, timeout: int = 30) -> Collection:
+        """Get the impact collection"""
+        response = requests.get(self.idu_events_collection_url, timeout=timeout)
+        if response.status_code == 200:
+            collection_dict = json.loads(response.text)
+            return Collection.from_dict(collection_dict)
+        return Collection.from_dict({})
+
+    def make_source_event_items(self) -> List[Item]:
+        """Create the source event item"""
+        items = []
+        idu_data = self.check_and_get_idu_data()
+        if not idu_data:
+            return []
+
+        for data in idu_data:
+            item = self.make_source_event_item(data=data)
+            items.append(item)
+        return items
+
+    def make_source_event_item(self, data: dict) -> Item:
+        """Create an Event Item"""
+        latitude = float(data.get("latitude"))
+        longitude = float(data.get("longitude"))
+        # Create the geojson point
+        point = Point(longitude, latitude)
+        geometry = mapping(point)
+        bbox = [longitude, latitude, longitude, latitude]
+
+        description = md(data["standard_popup_text"])
+
+        # Episode number not in the source, so, set it to 1
+        episode_number = 1
+
+        startdate_str = data["event_start_date"]
+        enddate_str = data["event_end_date"]
+
+        startdate = pytz.utc.localize(datetime.datetime.fromisoformat(startdate_str))
+        enddate = pytz.utc.localize(datetime.datetime.fromisoformat(enddate_str))
+
+        item = Item(
+            id=f'{STAC_EVENT_ID_PREFIX}{data["event_id"]}',
+            geometry=geometry,
+            bbox=bbox,
+            datetime=startdate,
+            properties={
+                "title": data["event_name"],
+                "description": description,
+                "start_datetime": startdate.isoformat(),
+                "end_datetime": enddate.isoformat(),
+                "location": data["locations_name"],
+                "sources": data["sources"],
+            },
+        )
+
+        item.set_collection(self.get_event_collection())
+        item.properties["roles"] = ["source", "event"]
+
+        item.add_asset("report", Asset(href=data["source_url"], media_type="application/pdf", title="Report"))
+        item.add_link(Link("via", self.data.get_source_url(), "application/json", "IDU Event Data"))
+
+        hazard_tuple = (data["category"], data["subcategory"], data["type"], data["subtype"])
+
+        MontyExtension.add_to(item)
+        monty = MontyExtension.ext(item)
+        monty.episode_number = episode_number
+        monty.country_codes = [data["iso3"]]
+        monty.hazard_codes = self.map_idu_to_hazard_codes(hazard=hazard_tuple)
+        monty.compute_and_set_correlation_id(hazard_profiles=self.hazard_profiles)
+
+        return item
+
+    def map_idu_to_hazard_codes(self, hazard: tuple):
```

Comment:
IDMC uses the [EM-DAT CRED hazard taxonomy](https://github.com/IFRCGo/monty-stac-extension/blob/a696fe30cc6cd98a8af0b236307fcd36c8a45ff3/model/sources/IDMC/README.md#hazard-type-mapping), please update the method accordingly.

Reply:
@emmanuelmathot I created the mapping based on the above taxonomy. From the source we get the category, sub-category, type, sub-type. Each combination is then mapped to the Hazard Code based on that taxonomy.

I understood that in EM_DAT, we get the classification_key from source which you have mapped to the category, subcategory, type, sub-type.

Let me know if you are thinking it differently. If you could give a small example, that would be helpful.

Reply:
this is the UNDRR classificiation you are using. We need to use the [EM-DAT CRED hazard taxonomy](https://github.com/IFRCGo/monty-stac-extension/blob/a696fe30cc6cd98a8af0b236307fcd36c8a45ff3/model/taxonomy.md#em-dat-cred-classification-tree). No worry, I will make the change with a common helper.